cmd/crates/soroban-spec-typescript/src/lib.rs

```diff
@@ -121,14 +121,28 @@ export class Client extends ContractClient {{
 }
 
 pub fn generate(spec: &[ScSpecEntry]) -> String {
-    let mut collected: Vec<_> = spec.iter().map(Entry::from).collect();
-    if !spec.iter().any(is_error_enum) {
-        collected.push(Entry::ErrorEnum {
-            doc: String::new(),
-            name: "Error".to_string(),
-            cases: vec![],
-        });
-    }
+    let collected: Vec<_> = spec.iter().map(Entry::from).collect();
+    let (error_docs, error_cases) = collected
+        .clone()
+        .into_iter()
+        .filter_map(|entry| {
+            if let Entry::ErrorEnum { doc, cases, .. } = entry {
+                Some((doc, error_cases_to_ts(&cases)))
+            } else {
+                None
+            }
+        })
+        .fold(
+            (String::new(), vec![]),
+            |(acc_docs, acc_cases), (docs, cases)| {
+                (format!("{acc_docs}{docs}"), [acc_cases, cases].concat())
+            },
+        );
+    let error_docs = doc_to_ts_doc(&error_docs, None, 0);
+    let errors = format!(
+        "{error_docs}export const Errors = {{\n{}\n}}",
+        error_cases.join(",\n")
+    );
```

Comment:
A raw string is more readable IMHO
```suggestion
    let errors = format!(
        r#"{error_docs}export const Errors = {{
{}
}}"#,
        error_cases.join(",\n")
    );
```

Reply:
Disagree, don't like the multiline!

Reply:
I try to always do it. I hate seeing `\n` in a string literal if it's a template.

Reply:
Sure, we'll still have some `\n` in there. It's a nice one-liner, though. The raw string is significantly uglier imho.